src/main/java/io/wisetime/connector/utils/TagDurationCalculator.java

```diff
@@ -30,18 +31,33 @@ public static double tagDuration(final TimeGroup timeGroup) {
    * @return duration, in seconds, to apply to each tag
    */
   public static double tagDurationDisregardingExperienceRating(final TimeGroup timeGroup) {
-    return tagDuration(timeGroup, true);
+    return tagDuration(timeGroup, true, true);
   }
 
-  private static double tagDuration(final TimeGroup timeGroup, final boolean disregardExperienceRating) {
+
+  /**
+   * Calculate the tag duration to apply to each tag.
+   *
+   * @param timeGroup the TimeGroup for which to calculate each tag duration
+   * @param useTimeGroupDuration if True, the method will use the time group's total duration to calculate tag duration,
+   *                             otherwise, it will use the sum of the duration of each time row in the time group.
+   * @param disregardExperienceRating if True, it will apply the user experience rating to calculate tag duration, otherwise,
+   *                                  it will disregard user's experience rating.
+   * @return duration, in seconds, to apply to each tag
+   */
+  public static double tagDuration(final TimeGroup timeGroup,
```

Comment:
Patricia uses the total duration of group time rows to record actual hours worked in Time Registration